### PR TITLE
[ISSUE #4745]Check controllerDLegerPeers and controllerDLegerSelfId attributes before create DLedgerServer

### LINKED ISSUE #4745: DLedgerController create DLedgerServer not check controllerDLegerPeers and controllerDLegerSelfId result in throw NullPointerException
When start Controller，only set ROCKETMQ_HOME, as following pictrue:
image
then project throws a  **java.lang.NullPointerException** :
image
why? Because ControllerConfig class attribute **controllerDLegerPeers**  is null, but this attribute cannot be null. in ControllerConfig class has three attributes:

- controllerDLegerGroup
- controllerDLegerPeers
- controllerDLegerSelfId
if not config in file,  they will be null. when we create DLedgerServer controllerDLegerPeers and controllerDLegerSelfId  must have a value. so when we create DLedgerServer  must check.
image

### PR DESCRIPTION
**Pull request by mxsm**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

close #4745 

## Brief changelog

- Check controllerDLegerPeers and controllerDLegerSelfId attributes before create DLedgerServer

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Review by RongtongJin — COMMENTED** [review]
[ERROR] /home/travis/build/apache/rocketmq/controller/src/main/java/org/apache/rocketmq/controller/ControllerManager.java:97:9: 'if' is not followed by whitespace. [WhitespaceAround]
[ERROR] /home/travis/build/apache/rocketmq/controller/src/main/java/org/apache/rocketmq/controller/ControllerManager.java:97:83: '{' is not preceded with whitespace. [WhitespaceAround]

Some check style failed

File: controller/src/main/java/org/apache/rocketmq/controller/ControllerManager.java [hunk 1]
```diff
@@ -91,6 +91,12 @@ protected <T> RunnableFuture<T> newTaskFor(final Runnable runnable, final T valu
             }
         };
         this.heartbeatManager = new DefaultBrokerHeartbeatManager(this.controllerConfig);
+        if (StringUtils.isEmpty(this.controllerConfig.getControllerDLegerPeers())) {
+            throw new IllegalArgumentException("Attribute value controllerDLegerPeers of ControllerConfig is null");
+        }
+        if(StringUtils.isEmpty(this.controllerConfig.getControllerDLegerSelfId())){
+            throw new IllegalArgumentException("Attribute value controllerDLegerSelfId of ControllerConfig is null");
```
Comment (caigy) [inline controller/src/main/java/org/apache/rocketmq/controller/ControllerManager.java:98]: 'is null or empty' may be more accurate.

**Review by caigy — CHANGES_REQUESTED** [review]

File: controller/src/main/java/org/apache/rocketmq/controller/ControllerManager.java (hunk 1 above)
Comment (mxsm) [reply controller/src/main/java/org/apache/rocketmq/controller/ControllerManager.java:98]: > 'is null or empty' may be more accurate.

OK I will modify it

**Comment by mxsm** [comment]
> [ERROR] /home/travis/build/apache/rocketmq/controller/src/main/java/org/apache/rocketmq/controller/ControllerManager.java:97:9: 'if' is not followed by whitespace. [WhitespaceAround] [ERROR] /home/travis/build/apache/rocketmq/controller/src/main/java/org/apache/rocketmq/controller/ControllerManager.java:97:83: '{' is not preceded with whitespace. [WhitespaceAround]
> 
> Some check style failed

I resubmitted the code

**Review by RongtongJin — APPROVED** [review]

**Pull request merged**